include/tokenizers_cpp.h

```diff
@@ -73,6 +84,38 @@ class Tokenizer {
    * \return The created tokenizer.
    */
   static std::unique_ptr<Tokenizer> FromBlobRWKVWorld(const std::string& model_blob);
+
+  /*!
+   * \brief Returns the vocabulary size.
+   */
+  virtual size_t GetVocabSize() = 0;
+
+  /*!
+   * \brief Returns the vocabulary as a dictionary of string token to index.
+   */
+  Vocab GetVocab() { return vocab_; }
+
+  /*!
+   * \brief Returns the mapping from index to string token.
+   */
+  IdToToken GetIdToToken() { return id_to_token_; }
+
+ protected:
+  // Mapping from token str to id
+  Vocab vocab_;
+  // Mapping from id to token str
+  IdToToken id_to_token_;
+
+  // Build the vocab and id_to_token using GetVocabSize() and Decode().
+  // Used to assist the logic in constructor.
+  void BuildVocab() {
+    auto vocab_size = GetVocabSize();
+    for (int i = 0; i < static_cast<int>(vocab_size); ++i) {
+      auto token = Decode({i});
+      vocab_[token] = i;
+      id_to_token_[i] = token;
```

Comment:
we should directly expose id_to_token in the tokenizer, instead of leveraging decode, which can cost some time. Instead, directly expose tokenizer->IdToToken